### PR TITLE
chore(lint): Run prettier on most files and move ABIs to own file

### PR DESCRIPTION
**Pull request by CoreyMartin**
- ABIs are difficult to conform to prettier and are referenced in multiple files so they've been moved to a prettier ignored file
- Run prettier on all non conforming files
- Add `prettier --check .` as a test requirement in CI so that PRs must conform before merge

### CODE REVIEW

**Comment by jasonstockman** [comment]
Was just thinking about this a few minutes ago! Moving ABI to their own file sounds good. For the prettifier we should run that for sure on everything at some point, but probably wait to do it once all pending PR are done, otherwise it will make merges tough. I'd like to also propose we `"printWidth": 120` in the prettier config so we have less forced line wraps in functions.

**Comment by CoreyMartin** [comment]
@jasonstockman Sounds good! Updated printWidth. I checked the merge status against CurveTests branch and it's actually not too bad, 3 file conflicts. I can totally wait to merge this until after yours is merged and fix the conflicts here. Since this PR is adding the check to the tests the files need to pass in order for the merge to happen.

File: MainnetHelper.js [hunk 1]
```diff
@@ -1,9 +1,24 @@
-const {BigNumber, FixedFormat, FixedNumber, formatFixed, parseFixed} = require("@ethersproject/bignumber");
+const {
```
Comment (yotamdaniel) [inline MainnetHelper.js:1]: Duplicate file?

File: test/VaultOUSD.js [hunk 2]
```diff
@@ -17,65 +17,107 @@ describe("VaultOUSD test suit", function () {
...
-        await MainnetHelper.helperSwapETHWithOUSD(addr1, ethers.utils.parseEther("1.0"))
-        await MainnetHelper.helperSwapETHWithOUSD(addr2, ethers.utils.parseEther("1.0"))
-        await MainnetHelper.helperSwapETHWithOUSD(owner, ethers.utils.parseEther("1.0"))
-        sharesOwnerAddress = owner.address
+        await MainnetHelper.helperSwapETHWithOUSD(
+            addr1,
+            ethers.utils.parseEther("1.0")
+        );
+        await MainnetHelper.helperSwapETHWithOUSD(
+            addr2,
```
Comment (yotamdaniel) [inline test/VaultOUSD.js:44]: can this be wrapped to a single line?

File: .github/workflows/BuildAndTest.yml [hunk 3]
```diff
@@ -4,49 +4,49 @@ name: Build and Test
...
-          npx hardhat test
+    # This workflow contains a single job called "build"
+    build:
+        # The type of runner that the job will run on
+        runs-on: ubuntu-latest
 
+        # Steps represent a sequence of tasks that will be executed as part of the job
+        steps:
+            # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+            - uses: actions/checkout@v2
```
Comment (jasonstockman) [inline .github/workflows/BuildAndTest.yml:26]: Should these also be v3?

**Review by jasonstockman — APPROVED** [review]

**Comment by jasonstockman** [comment]
> 

No need to wait on my changes. I can handle those once this is approved. The CurveTests branch wont be ready today and I'd rather get this in ASAP.

File: .github/workflows/BuildAndTest.yml (hunk 3 above)
Comment (CoreyMartin) [reply .github/workflows/BuildAndTest.yml:26]: yeah these are v3 after merging main back in

File: .github/workflows/BuildAndTest.yml [hunk 4]
```diff
@@ -4,49 +4,49 @@ name: Build and Test
...
+              with:
+                  submodules: "true"
+                  token: "${{ secrets.PAT }}"
+
+            - name: Setup NodeJS 16
```
[the remark this answers is not in the document]
Comment (CoreyMartin) [reply .github/workflows/BuildAndTest.yml:31]: yeah these are v3 after merging main back in

**Pull request merged**